api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirTaskDaoImpl.java

```diff
@@ -61,18 +60,18 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 		});
 	}
 	
-	@Override
-	public FhirTask createOrUpdate(FhirTask task) throws DAOException {
-		// TODO: Refactor - and figure out why CascadeType.ALL does not take care of this.
-		if (task.getOwnerReference() != null && task.getOwnerReference().getReference() != null) {
-			getSessionFactory().getCurrentSession().saveOrUpdate(task.getOwnerReference());
-		}
-		
-		getSessionFactory().getCurrentSession().saveOrUpdate(task);
-		
-		return task;
-	}
-	
```

Comment:
We probably want to keep this in to ensure the TaskOwnerReference is saved before the task is.

Reply:
ok thanks let me revert it back